### PR TITLE
[release 1.13] security: up deps in release branch

### PR DESCRIPTION
**Pull request by sicoyle**
# Description

I had issues cherry picking my merged commit directly from the main branch as it was bringing in a lot of other changes, so manually copy/pasted. For the cloudflare worker stuff I did up the dep and then recompile everything so that should be g2g.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: .build-tools/go.mod [hunk 1]
```diff
@@ -1,8 +1,8 @@
 module github.com/dapr/components-contrib/build-tools
 
-go 1.21
```
Comment (ItalyPaleAle) [inline .build-tools/go.mod:3]: Please don't change the Go version used for 1.13, as that's a major change

File: go.mod [hunk 2]
```diff
@@ -431,3 +434,8 @@ replace github.com/microcosm-cc/bluemonday => github.com/microcosm-cc/bluemonday
 
 // this is a fork which addresses a performance issues due to go routines.
 replace dubbo.apache.org/dubbo-go/v3 => dubbo.apache.org/dubbo-go/v3 v3.0.3-0.20230118042253-4f159a2b38f3
+
+// Uncomment for local development for testing with changes in the components-contrib && kit repositories.
+// Don't commit with this uncommented!
+//
+// replace github.com/dapr/kit => ../kit
```
Comment (ItalyPaleAle) [inline go.mod:441]: Why was this added to the release branch?

**Comment by sicoyle** [comment]
will revert the go version bump in a bit. Didn't realize that was in my changes as well, but already switched to dapr/dapr updates for this

File: go.mod [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
+	cloud.google.com/go/storage v1.36.0
 	dubbo.apache.org/dubbo-go/v3 v3.0.3-0.20230118042253-4f159a2b38f3
 	github.com/Azure/azure-sdk-for-go/sdk/ai/azopenai v0.3.0
-	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.9.1
-	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.5.1
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.11.1
+	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.5.2
 	github.com/Azure/azure-sdk-for-go/sdk/data/azappconfig v1.1.0
-	github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos v0.3.6
+	github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos v1.0.1
```
Comment (ItalyPaleAle) [inline go.mod:17]: Please revert this, this change is not a minor one as we go from v0.x to v1.x

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Ideally in a release branch we should never update dependencies unless they contain security issues or critical bug fixes. I am concerned with the fact that this PR updates A LOT of dependencies, including SDKs for cloud services.

File: go.mod (hunk 2 above)
Comment (sicoyle) [reply go.mod:441]: i just copy/pasted my changes from the other PR. Will come back and fix these bc didn't realize this also bumped go versions until i looked over my own pr. Give me a sec pls :)

**Comment by sicoyle** [comment]
> Ideally in a release branch we should never update dependencies unless they contain security issues or critical bug fixes. I am concerned with the fact that this PR updates A LOT of dependencies, including SDKs for cloud services.

These are all for security fixes based on my PR desc here: https://github.com/dapr/components-contrib/pull/3390

**Comment by ItalyPaleAle** [comment]
> > Ideally in a release branch we should never update dependencies unless they contain security issues or critical bug fixes. I am concerned with the fact that this PR updates A LOT of dependencies, including SDKs for cloud services.
> 
> These are all for security vulnerabilities based on my PR desc here: #3390

I see the `go.mod` contains a lot more deps updated than the ones you list in #3390, including SDKs for all cloud providers.

**Comment by sicoyle** [comment]
> I see the go.mod contains a lot more deps updated than the ones you list in https://github.com/dapr/components-contrib/pull/3390, including SDKs for all cloud providers.

I def did not do a `go get -u ...` on everything. Let me see what I can do. Still massaging everything 😁

**Comment by ItalyPaleAle** [comment]
> > I see the go.mod contains a lot more deps updated than the ones you list in #3390, including SDKs for all cloud providers.
> 
> I def did not do a `go get -u ...` on everything. Let me see what I can do. Part of this could also be that I had to copy/paste based on my other PR, so maybe just need to not grab less changes. I'm working on it 😁

Here's my tip:

1. Reset to `main`: `git fetch --all && get reset --hard origin/main`
2. Open the `go.mod` file (the main one) and update the deps that you listed in the table, and only those
3. `make modtidy-all`

**Comment by sicoyle** [comment]
flaky failure?
```
--- FAIL: Test_EndToEnd (0.19s)
    --- FAIL: Test_EndToEnd/local (0.06s)
        --- FAIL: Test_EndToEnd/local/consoleLog_multiple_requests (0.02s)
            e2e_test.go:118: 
                	Error Trace:	/home/runner/work/components-contrib/components-contrib/middleware/http/wasm/internal/e2e_test.go:118
                	            				/home/runner/work/components-contrib/components-contrib/middleware/http/wasm/internal/e2e_test.go:173
                	Error:      	"time=\"2024-06-26T16:07:13.331155155Z\" level=info msg=\"main ConsoleLog\" instance=fv-az1215-760 scope=Test_EndToEnd type=log ver=unknown\ntime=\"2024-06-26T16:07:13.33158788Z\" level=info msg=\"main ConsoleLog\" instance=fv-az1215-760 scope=Test_EndToEnd type=log ver=unknown\ntime=\"2024-06-26T16:07:13.331608729Z\" level=info msg=\"request[0] ConsoleLog\" instance=fv-az1215-760 scope=Test_EndToEnd type=log ver=unknown\ntime=\"2024-06-26T16:07:13.331616583Z\" level=debug msg=\"wasm stdout: main Stdout\\nmain Stdout\\nrequest[0] Stdout\\n\" instance=fv-az1215-760 scope=Test_EndToEnd type=log ver=unknown\ntime=\"2024-06-26T16:07:13.331621553Z\" level=debug msg=\"wasm stderr: main Stderr\\nmain Stderr\\nrequest[0] Stderr\\n\" instance=fv-az1215-760 scope=Test_EndToEnd type=log ver=unknown\ntime=\"2024-06-26T16:07:13.331631812Z\" level=info msg=\"request[1] ConsoleLog\" instance=fv-az1215-760 scope=Test_EndToEnd type=log ver=unknown\ntime=\"2024-06-26T16:07:13.33163606Z\" level=debug msg=\"wasm stdout: request[1] Stdout\\n\" instance=fv-az1215-760 scope=Test_EndToEnd type=log ver=unknown\ntime=\"2024-06-26T16:07:13.331639516Z\" level=debug msg=\"wasm stderr: request[1] Stderr\\n\" instance=fv-az1215-760 scope=Test_EndToEnd type=log ver=unknown\ntime=\"2024-06-26T16:07:13.331648322Z\" level=info msg=\"request[2] ConsoleLog\" instance=fv-az1215-760 scope=Test_EndToEnd type=log ver=unknown\ntime=\"2024-06-26T16:07:13.33165221Z\" level=debug msg=\"wasm stdout: request[2] Stdout\\n\" instance=fv-az1215-760 scope=Test_EndToEnd type=log ver=unknown\ntime=\"2024-06-26T16:07:13.331655425Z\" level=debug msg=\"wasm stderr: request[2] Stderr\\n\" instance=fv-az1215-760 scope=Test_EndToEnd type=log ver=unknown\n" does not contain "level=debug msg=\"wasm stdout: main Stdout\\nrequest[0] Stdout\\n\""
                	Test:       	Test_EndToEnd/local/consoleLog_multiple_requests
FAIL
```

File: go.mod [hunk 4]
```diff
@@ -134,11 +134,13 @@ require (
 	sigs.k8s.io/yaml v1.4.0
 )
 
+require github.com/felixge/httpsnoop v1.0.4 // indirect
+
 require (
```
Comment (ItalyPaleAle) [inline go.mod:139]: Single block please!

File: go.mod [hunk 5]
```diff
@@ -6,9 +6,9 @@ toolchain go1.21.4
 
 require (
 	cloud.google.com/go/datastore v1.15.0
-	cloud.google.com/go/pubsub v1.33.0
-	cloud.google.com/go/secretmanager v1.11.2
-	cloud.google.com/go/storage v1.33.0
+	cloud.google.com/go/pubsub v1.36.1
```
Comment (ItalyPaleAle) [inline go.mod:9]: Why was the Google Cloud SDK updated? This was not in the table with the list of vulnerabilities

File: go.mod [hunk 6]
```diff
@@ -26,7 +26,7 @@ require (
 	github.com/Azure/azure-sdk-for-go/sdk/storage/azqueue v1.0.0
 	github.com/Azure/go-amqp v1.0.5-0.20240301200753-2dff4b36f85f
 	github.com/DATA-DOG/go-sqlmock v1.5.0
-	github.com/IBM/sarama v1.42.2
+	github.com/IBM/sarama v1.42.1
```
Comment (ItalyPaleAle) [inline go.mod:29]: Downgraded dependency?

File: go.mod [hunk 7]
```diff
@@ -65,10 +65,10 @@ require (
 	github.com/gocql/gocql v1.5.2
 	github.com/golang/mock v1.6.0
 	github.com/google/uuid v1.6.0
-	github.com/googleapis/gax-go/v2 v2.12.0
+	github.com/googleapis/gax-go/v2 v2.12.2
 	github.com/gorilla/mux v1.8.1
 	github.com/grandcat/zeroconf v1.0.0
-	github.com/hamba/avro/v2 v2.15.0
+	github.com/hamba/avro/v2 v2.21.1
```
Comment (ItalyPaleAle) [inline go.mod:71]: According to your table, the vulnerability was in < 2.13.0. We were using 2.15.0 so should have been safe. No need to update this dependency then: in a release branch, it's best to be as conservative as possible.

File: go.mod [hunk 8]
```diff
@@ -114,14 +114,14 @@ require (
...
-	google.golang.org/api v0.138.0
-	google.golang.org/grpc v1.59.0
-	google.golang.org/protobuf v1.32.0
+	golang.org/x/crypto v0.24.0
+	golang.org/x/exp v0.0.0-20231006140011-7918f672742d
+	golang.org/x/mod v0.17.0
+	golang.org/x/net v0.25.0
+	golang.org/x/oauth2 v0.18.0
+	google.golang.org/api v0.169.0
+	google.golang.org/grpc v1.64.0
```
Comment (ItalyPaleAle) [inline go.mod:123]: Your table shows that the patch was in 1.56.3. We were using 1.59.0 so no need to upgrade the release branch. This is a critical dependency for Dapr.

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: go.mod (hunk 5 above)
Comment (sicoyle) [reply go.mod:9]: this is required due to bumping the grpc package to the version without a security vulnerability. I just tried downgrading it back and it bumps us down for grpc pkg unfortunately

File: go.mod (hunk 8 above)
Comment (sicoyle) [reply go.mod:123]: I see. How about I bump it to match what runtime has for grpc for the release 1.13 branch. See here https://github.com/dapr/dapr/blob/release-1.13/go.mod
`[google.golang.org/grpc](http://google.golang.org/grpc) v1.60.1`

Comment (sicoyle) [reply go.mod:123]: thoughts @ItalyPaleAle ?

Comment (ItalyPaleAle) [reply go.mod:123]: Ok

File: go.mod (hunk 5 above)
Comment (ItalyPaleAle) [reply go.mod:9]: See if using 1.60.1 requires bumping too, see other thread

**Comment by sicoyle** [comment]
@ItalyPaleAle are these transient failures in the build..? I think it's ready ready ready, but yeah need help knowing if these failures are flaky or not 🚀 🤓 ⭐

File: tests/e2e/pubsub/jetstream/go.mod [hunk 9]
```diff
@@ -1,23 +1,23 @@
 module github.com/dapr/components-contrib/tests/e2e/pubsub/jetstream
 
-go 1.21
+go 1.22.0
```
Comment (ItalyPaleAle) [inline tests/e2e/pubsub/jetstream/go.mod:3]: Please no changing Go version

File: tests/certification/state/azure/cosmosdb/cosmosdb_test.go [hunk 10]
```diff
@@ -133,14 +133,17 @@ func TestAzureCosmosDBStorage(t *testing.T) {
 				"partitionKey": "mypartition",
 			}
 
-			test := func(setMeta, getMeta map[string]string, expectedValue string) {
+			test := func(setMeta, getMeta map[string]string, expectedValue string, expectedErr bool) {
```
Comment (ItalyPaleAle) [inline tests/certification/state/azure/cosmosdb/cosmosdb_test.go:136]: Why these changes?

File: tests/certification/secretstores/hashicorp/vault/metadata_helpers.go [hunk 11]
```diff
@@ -74,7 +74,7 @@ func getComponentCapabilities(ctx flow.Context, currentGrpcPort int, targetCompo
 
 	clientCtx := context.Background()
 
-	resp, err := daprClient.GrpcClient().GetMetadata(clientCtx, &runtime.GetMetadataRequest{})
+	resp, err := daprClient.GrpcClient().GetMetadata(clientCtx, new(emptypb.Empty))
```
Comment (ItalyPaleAle) [inline tests/certification/secretstores/hashicorp/vault/metadata_helpers.go:77]: This change doesn't seem right?

File: tests/certification/go.mod [hunk 12]
```diff
@@ -19,32 +19,33 @@ require (
 	github.com/cenkalti/backoff/v4 v4.2.1
 	github.com/cloudwego/kitex v0.5.0
 	github.com/cloudwego/kitex-examples v0.1.1
-	github.com/dapr/components-contrib v1.13.0-rc.6
-	github.com/dapr/dapr v1.13.0-rc.7
-	github.com/dapr/go-sdk v1.6.1-0.20231102031149-87bbb8cd690a
-	github.com/dapr/kit v0.13.0
+	github.com/dapr/components-contrib v1.13.1
+	github.com/dapr/dapr v1.13.4
+	github.com/dapr/go-sdk v1.6.0
```
Comment (ItalyPaleAle) [inline tests/certification/go.mod:24]: This may be the reason for the change that "does not seem right"... you've downgraded the go-sdk package

File: tests/certification/configuration/redis/redis_test.go [hunk 13]
```diff
@@ -292,7 +292,7 @@ func TestRedis(t *testing.T) {
 			client := sidecar.GetClient(ctx, sidecarName1)
 			message.Reset()
 			var errSubscribe error
-			subscribeID, errSubscribe = client.SubscribeConfigurationItems(ctx, storeName, keys, func(id string, items map[string]*dapr.ConfigurationItem) {
+			errSubscribe = client.SubscribeConfigurationItems(ctx, storeName, keys, func(id string, items map[string]*dapr.ConfigurationItem) {
```
Comment (ItalyPaleAle) [inline tests/certification/configuration/redis/redis_test.go:295]: this doesn't seem right either

File: tests/certification/configuration/postgres/postgres_test.go [hunk 14]
```diff
@@ -202,7 +202,7 @@ func TestPostgres(t *testing.T) {
 		return func(ctx flow.Context) error {
 			client := sidecar.GetClient(ctx, sidecarName)
 			message.Reset()
-			subscribeID, errSubscribe := client.SubscribeConfigurationItems(ctx, storeName, keys, func(id string, items map[string]*dapr.ConfigurationItem) {
+			errSubscribe := client.SubscribeConfigurationItems(ctx, storeName, keys, func(id string, items map[string]*dapr.ConfigurationItem) {
```
Comment (ItalyPaleAle) [inline tests/certification/configuration/postgres/postgres_test.go:205]: Ditto

File: state/azure/cosmosdb/cosmosdb_query.go [hunk 15]
```diff
@@ -286,6 +286,7 @@ func (q *Query) execute(ctx context.Context, client *azcosmos.ContainerClient) (
 		}
 
 		token = queryResponse.ContinuationToken
+
```
Comment (ItalyPaleAle) [inline state/azure/cosmosdb/cosmosdb_query.go:289]:
```suggestion
```

File: go.mod (hunk 4 above)
Comment (ItalyPaleAle) [inline go.mod:139]: Same block please

File: common/component/cloudflare/worker-src/package.json [hunk 16]
```diff
@@ -2,7 +2,7 @@
     "private": true,
     "name": "dapr-cfworkers-client",
     "description": "Client code for Dapr to interact with Cloudflare Workers",
-    "version": "20230517",
+    "version": "20230625",
```
Comment (ItalyPaleAle) [inline common/component/cloudflare/worker-src/package.json:5]:
```suggestion
    "version": "20240625",
```

:)

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: tests/certification/secretstores/hashicorp/vault/metadata_helpers.go (hunk 11 above)
Comment (sicoyle) [reply tests/certification/secretstores/hashicorp/vault/metadata_helpers.go:77]: this was fixing a linter issue. If I leave it as is with old val for the param, then linter is not happy.

File: tests/certification/state/azure/cosmosdb/cosmosdb_test.go (hunk 10 above)
Comment (sicoyle) [reply tests/certification/state/azure/cosmosdb/cosmosdb_test.go]: this was copy/paste from me looking at other prs to bring in changes for func signatures and copying this too. It's not needed so reverted.

**Comment by sicoyle** [comment]
pls note that make modtidy-all with specifying my go version for go 1.21 creates two require blocks in the tests/certification go mod. That needs to stay to keep linter happy. I think I've addressed everything :)

File: tests/certification/secretstores/hashicorp/vault/metadata_helpers.go (hunk 11 above)
Comment (ItalyPaleAle) [reply tests/certification/secretstores/hashicorp/vault/metadata_helpers.go:77]: Not a linter issue. Per my other comment, this was due to the fact that you had downgraded the Dapr Go SDK. After you've un-downgraded that, the test now fails to build: https://github.com/dapr/components-contrib/actions/runs/9687303478/job/26731660485
Please revert this change and it should work

**Comment by sicoyle** [comment]
anything else @ItalyPaleAle or g2g? :)

File: common/component/cloudflare/worker-src/package.json (hunk 16 above)
Comment (ItalyPaleAle) [reply common/component/cloudflare/worker-src/package.json:5]: This has not been resolved. The date is 1 year off...

**Review by ItalyPaleAle — APPROVED** [review]
